drivers/network/virtio/ethernet.c

```diff
@@ -109,12 +109,12 @@ static void rx_provide(void)
             assert(!err);
 
             // Allocate a desc entry for the header, and one for the packet
-            uint32_t hdr_desc_idx;
+            uint32_t hdr_desc_idx = -1;
             err = ialloc_alloc(&rx_ialloc_desc, &hdr_desc_idx);
-            assert(!err);
-            uint32_t pkt_desc_idx;
+            assert(!err && hdr_desc_idx != -1);
```

Comment:
Is it guaranteed that ialloc_alloc can't set hdr_desc_idx to 0xFFFFFFFF validly?

Reply:
The extra assert check is probably overkill because we have an invariant that `ialloc_alloc` always succeeds, the extra assert condition just makes sure nothing has gone wrong in `ialloc_alloc` and we actually do get a proper index.